docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+   :language: c
+   :start-after: start getCharPtr1
+   :end-before: end getCharPtr1
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: const char * getCharPtr1()
+
+The C wrapper copies all of the metadata into a ``SHROUD_array``
+struct which is used by the Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/strings/wrapstrings.cpp
+   :language: c
+   :start-after: start STR_get_char_ptr1_bufferify
+   :end-before: end STR_get_char_ptr1_bufferify
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start c_get_char_ptr1_bufferify
+   :end-before: end c_get_char_ptr1_bufferify
+   :dedent: 8
+
+The Fortran wrapper uses the metadata in ``DSHF_rv`` to allocate
+a ``CHARACTER`` variable of the correct length.
+The helper function ``SHROUD_copy_string_and_free`` will copy 
+the results of the C++ function into the return variable:
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start get_char_ptr1
+   :end-before: end get_char_ptr1
+   :dedent: 4
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    character(len=:), allocatable :: str
+    str = get_char_ptr1()
+
+.. ############################################################
+
+.. _example_getCharPtr2:
+
+getCharPtr2
+^^^^^^^^^^^
+
+If you know the maximum size of string that you expect the function to
+return, then the *len* attribute is used to declare the length.  The
+explicit ``ALLOCATE`` is avoided but any result which is longer than
+the length will be silently truncated.
+
+.. literalinclude:: ../regression/run/strings/strings.cpp
+   :language: c
+   :start-after: start getCharPtr2
+   :end-before: end getCharPtr2
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: const char * getCharPtr2() +len(30)
+
+The C wrapper:
+
+.. literalinclude:: ../regression/reference/strings/wrapstrings.cpp
+   :language: c
+   :start-after: start STR_get_char_ptr2_bufferify
+   :end-before: end STR_get_char_ptr2_bufferify
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start c_get_char_ptr2_bufferify
+   :end-before: end c_get_char_ptr2_bufferify
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start get_char_ptr2
+   :end-before: end get_char_ptr2
+   :dedent: 4
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    character(30) str
+    str = get_char_ptr2()
+
+.. ############################################################
+
+.. _example_getCharPtr3:
+
+getCharPtr3
+^^^^^^^^^^^
+
+Create a Fortran subroutine in an additional ``CHARACTER``
+argument for the C function result. Any size character string can
+be returned limited by the size of the Fortran argument.  The
+argument is defined by the *F_string_result_as_arg* format string.
+Works with Fortran 90.
```

Comment:
>Fortran 90

Please clarify.  Does this work with "Fortran 90, or later", or only with "Fortran 90"?

Reply:
The intent was to make the point that allocating a character string to a specific length is in 2003 and this was intended as a work around for older compilers. However, there are too many dependencies on other 2003 features so I removed the comment.